docs/docs/08-guides/03-react-vite-websockets.md

```diff
@@ -0,0 +1,492 @@
+
+## Step 2 - Hello `@winglibs/vite` 
+
+In the previous step, we used `npm run dev` to start the local web server. 
+In this step, we will install the `@winglibs/vite` package responsible for starting the dev server. 
+We will also pass information from the backend to the frontend.
+
+### Install and use `@winglibs/vite`
+
+1. Install `@winglibs/vite`:
+ ```sh
+ cd ~/shared-counter/backend
+ npm i -s @winglibs/vite
+ ```
+1. Bring and instantiate Vite in `backend/main.w`:
+ ```wing
+ bring vite;
+ 
+ new vite.Vite(
+   root:"../frontend"
+ );
+ ```
+3. Run this project inside the Wing Simulator:
+ ```sh
+ wing run main.w
+ ```
+> You should have 2 web pages open: the browser and the Wing Console, showing the simulator.
+
+### Sending data to your Vite app using `publicEnv`
+
+Now that we have our backend instantiate the Vite resource, 
+we would like to see how we can pass constant data from the backend to the frontend.
+
+1. Use `publicEnv` for the title in `backend/main.w:
+ ```wing
+ bring vite;
+ 
+ new vite.Vite(
+   root:"../frontend",
+   publicEnv: {
+     title: "Wing + Vite + React"
+   }
+ );
+ ```
+2. Notice that the React webpage now has `window.wing.env` containing this title. 
+You can verify it by opening javascript console under developer tools and running `console.log(window.wing.env);`
+4. Read `window.wing.env.title` in `frontend/src/App.tsx` by looking for `<h1>Vite + React</h1>` and replacing it with `<h1>{window.wing.env.title}</h1>`.
+5. Upon saving both the Wing file and the TypeScript file, you should see the new title.
+
+## Step 3 - Adding a counter
+
+Now that we understand how to pass information from the backend to the frontend, we will create a backend API endpoint and provide the frontend code with a URL.
+On the frontend, we will switch from using a local counter to a backend-based counter.
+
+### Creating a counter and read/update API routes
+
+1. Instantiate a `cloud.Api` in `backend/main.w` by adding the following code:
+```wing
+bring vite;
+bring cloud;
+
+let api = new cloud.Api(cors: true);
+
+new vite.Vite(
+  root: "../frontend",
+  publicEnv: {
+    title: "Wing + Vite + React",
+    API_URL: api.url
+  }
+);
+```
+Notice that we've a new environment variable called `API_URL` which points to the URL of our API endpoint.
+
+2. Now, let's also instantiate a `cloud.Counter`:
+ 
+```wing
+let counter = new cloud.Counter();
+``` 
+3. Add the following routes:
+  - A `GET /counter` for retrieving the counter value (using `counter.peek()`)
+  ```wing
+  api.get("/counter", inflight () => {
+    return {
+      status: 200, 
+      body: "{counter.peek()}"
+    };
+  });
+  ```
+  - A `POST /counter` for incrementing the counter (using `counter.inc()`)
+  ```wing
+  api.post("/counter", inflight () => {
+    let oldValue = counter.inc();
+    return {
+      status: 200, 
+      body: "{oldValue + 1}"
+    };
+  });
+  ```
+4. Experiment with the Wing Simulator to see that these routes work as expected.
```

Comment:
you are using the terms Wing Console and Wing Simulator for the same product... I think you should choose one.